centreon/www/include/configuration/configObject/contact/formContact.php

```diff
@@ -479,6 +479,147 @@
     );
 }
 
+/* ------------------------ Topoogy ---------------------------- */
+$pages = [null => ""];
+$aclUser = $centreon->user->lcaTStr;
+if (! empty($aclUser)) {
+    $acls = array_flip(explode(',', $aclUser));
+    /**
+     * Transform [1, 2, 101, 202, 10101, 20201] to :
+     *
+     * 1
+     *   101
+     *     10101
+     * 2
+     *   202
+     *     20201
+     */
+    $createTopologyTree = function (array $topologies): array {
+        ksort($topologies, \SORT_ASC);
+        $parentsLvl = [];
+
+        // Classify topologies by parents
+        foreach (array_keys($topologies) as $page) {
+            if (strlen($page) == 1) {
+                // MENU level 1
+                if (! array_key_exists($page, $parentsLvl)) {
+                    $parentsLvl[$page] = [];
+                }
+            } elseif (strlen($page) == 3) {
+                // MENU level 2
+                $parentLvl1 = substr($page, 0, 1);
+                if (! array_key_exists($parentLvl1, $parentsLvl)) {
+                    $parentsLvl[$parentLvl1] = [];
+                }
+                if (! array_key_exists($page, $parentsLvl[$parentLvl1])) {
+                    $parentsLvl[$parentLvl1][$page] = [];
+                }
+            } elseif (strlen($page) == 5) {
+                // MENU level 3
+                $parentLvl1 = substr($page, 0, 1);
+                $parentLvl2 = substr($page, 0, 3);
+                if (! array_key_exists($parentLvl1, $parentsLvl)) {
+                    $parentsLvl[$parentLvl1] = [];
+                }
+                if (! array_key_exists($parentLvl2, $parentsLvl[$parentLvl1])) {
+                    $parentsLvl[$parentLvl1][$parentLvl2] = [];
+                }
+                if (! in_array($page, $parentsLvl[$parentLvl1][$parentLvl2])) {
+                    $parentsLvl[$parentLvl1][$parentLvl2][] = $page;
+                }
+            }
+        }
+
+        return $parentsLvl;
+    };
+
+    /**
+     * Check if at least one child can be shown
+     */
+    $oneChildCanBeShown = function () use (&$childrenLvl3, &$translatedPages): bool {
+        $isCanBeShow = false;
+        foreach ($childrenLvl3 as $topologyPage) {
+            if ($translatedPages[$topologyPage]['show']) {
+                $isCanBeShow = true;
+                break;
+            }
+        }
+        return $isCanBeShow;
```

Comment:
Change proposition to avoid useless variable.
```suggestion
        foreach ($childrenLvl3 as $topologyPage) {
            if ($translatedPages[$topologyPage]['show']) {
                return true;
            }
        }
        return false;
```